beacon/sync/src/test/java/tech/pegasys/teku/beacon/sync/historical/ProgressLoggerTest.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright ConsenSys Software Inc., 2022
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.beacon.sync.historical;
+
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import org.hyperledger.besu.metrics.noop.NoOpMetricsSystem;
+import org.hyperledger.besu.plugin.services.MetricsSystem;
+import org.junit.jupiter.api.Test;
+import tech.pegasys.teku.infrastructure.logging.StatusLogger;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.TestSpecFactory;
+import tech.pegasys.teku.spec.datastructures.blocks.SignedBeaconBlock;
+import tech.pegasys.teku.spec.util.DataStructureUtil;
+
+public class ProgressLoggerTest {
+  private ProgressLogger progressLogger;
+  private final MetricsSystem metricsSystem = new NoOpMetricsSystem();
+  private final StatusLogger statusLogger = mock(StatusLogger.class);
+
+  protected Spec spec = TestSpecFactory.createMinimalPhase0();
+  protected DataStructureUtil dataStructureUtil = new DataStructureUtil(spec);
+
+  @Test
+  public void shouldLogUpdatedStatusAfter5Minutes() {
+    final Instant lastLogged = Instant.now().minus(6, ChronoUnit.MINUTES);
+    progressLogger = new ProgressLogger(metricsSystem, statusLogger, lastLogged);
```

Comment:
Usually with time etc we just use `TimeProvider` and `StubTimeProvider` 

This would allow a stub time to be progressed as much as you like in the test case, and you could walk forward to any time you like...